### PR TITLE
infra: add github action for stale issues

### PR DESCRIPTION
**Pull request by avawang1**
*Issue #, if available:*

*Description of changes:*
* add github action for closing stale issues

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [X] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [X] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [X] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/stale_issue.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+          days-before-ancient: 1095
+
+          # If you don't want to mark a issue as being ancient based on a
+          # threshold of "upvotes", you can set this here. An "upvote" is
+          # the total number of +1, heart, hooray, and rocket reactions
+          # on an issue.
+          minimum-upvotes-to-exempt: 1
+
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+          loglevel: DEBUG
```
Comment (speller26) [inline .github/workflows/stale_issue.yml:45]: vs `INFO`?

File: .github/workflows/stale_issue.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        with:
+          # Setting messages to an empty string will cause the automation to skip
+          # that category
+          ancient-issue-message: Greetings! It looks like this issue hasn’t been active in longer than three years. We encourage you to check if this is still an issue in the latest release. Because it has been longer than three years since the last update on this, and in the absence of more information, we will be closing this issue soon. If you find that this is still a problem, please feel free to provide a comment to prevent automatic closure, or if the issue is already closed, please feel free to reopen it.
+          stale-issue-message: Greetings! It looks like this issue hasn’t been active in longer than a week. We encourage you to check if this is still an issue in the latest release. Because it has been longer than a week since the last update on this, and in the absence of more information, we will be closing this issue soon. If you find that this is still a problem, please feel free to provide a comment or add an upvote to prevent automatic closure, or if the issue is already closed, please feel free to open a new one.
```
Comment (speller26) [inline .github/workflows/stale_issue.yml:19]: What distinguishes ancient from stale? These messages are the same aside from the timeline?

File: .github/workflows/stale_issue.yml (hunk 1 above)
Comment (avawang1) [reply .github/workflows/stale_issue.yml:45]: This will just be in the logs for the github action

File: .github/workflows/stale_issue.yml (hunk 2 above)
Comment (avawang1) [reply .github/workflows/stale_issue.yml:19]: Yeah, ancient is for a longer time frame than stale and ancient also allows for exemption based on input minimum number of votes, see line 42. We can adjust as needed as we get more issues.

File: .github/workflows/stale_issue.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+          closed-for-staleness-label: closed-for-staleness
+
+          # Issue timing
+          days-before-stale: 7
+          days-before-close: 4
+          days-before-ancient: 1095
+
+          # If you don't want to mark a issue as being ancient based on a
+          # threshold of "upvotes", you can set this here. An "upvote" is
+          # the total number of +1, heart, hooray, and rocket reactions
```
Comment (speller26) [inline .github/workflows/stale_issue.yml:40]: Hah, that's arbitrary

**Pull request merged**